omigami/tests/test_spectra_matching.py

```diff
@@ -0,0 +1,148 @@
+                    }
+                ],
+            }
+        }
+    }
+
+    with pytest.raises(InvalidCredentials):
+        client._send_request(small_payload, spec2vec_prediction_endpoints["positive"])
+
+
+def test_format_results(sample_response):
+    client = SpectraMatching()
+    requests.Response()
+
+    results = client._format_results(sample_response)
+
+    assert isinstance(results[0], pd.DataFrame)
+    assert results[0].index.name == "matches of spectrum-0"
+    assert all(results[0]["score"] > 0)
+    assert all(results[0].columns == ["score", "compound_name", "smiles"])
+
+
+def test_validate_parameters():
+    parameters = SpectraMatching._build_parameters(2)
+    assert {"n_best_spectra"} == set(parameters.keys())
+
+    with pytest.raises(ValueError, match="must be an integer"):
+        _ = SpectraMatching._build_parameters("robin")
+
+
+def test_validate_input():
+    model_input = {
+        "peaks_json": "[[80.060677, 157.0], [337.508301, 230.0]]",
+        "Precursor_MZ": "153.233",
+    }
+    client = SpectraMatching()
+    # first validates if the input is correct then we test for errors
+    client._validate_input([model_input])
+
+    with pytest.raises(TypeError, match="Spectrum data must be a dictionary."):
+        client._validate_input(["not_a_dict"])
+
+    with pytest.raises(KeyError, match="mandatory keys"):
+        client._validate_input([{"Precursor_MZ": "1", "peaks_JASON": "[not a list]"}])
+
+    with pytest.raises(
+        ValueError, match="peaks_json needs to be a valid python string representation"
+    ):
+        client._validate_input([{"Precursor_MZ": "1", "peaks_json": "[not a list]"}])
+
+    with pytest.raises(
+        ValueError,
+        match="peaks_json needs to be a valid python string representation",
+    ):
+        client._validate_input([{"Precursor_MZ": "1", "peaks_json": 10}])
+
+    with pytest.raises(
+        ValueError,
+        match="Precursor_MZ needs to be a string representation of a float",
+    ):
+        client._validate_input([{"Precursor_MZ": "float", "peaks_json": [10]}])
+
+
+def test_create_spectra_generator_file_source(small_mgf_path):
+    client = SpectraMatching()
+
+    spectra_generator = client._create_spectra_generator(small_mgf_path)
+
+    assert isinstance(spectra_generator.__next__(), Spectrum)
+
+
+def test_create_spectra_generator_stream_source(small_mgf_path):
+    client = SpectraMatching()
+    stream = StringIO(open(small_mgf_path, "r").read())
+
+    spectra_generator = client._create_spectra_generator(stream)
+
+    assert isinstance(spectra_generator.__next__(), Spectrum)
+
+
+def test_create_spectra_generator_spectrum_source(small_mgf_path):
+    client = SpectraMatching()
+    spectra = list(load_from_mgf(small_mgf_path))
+
+    spectra_generator = client._create_spectra_generator(spectra)
+
+    assert isinstance(spectra_generator.__next__(), Spectrum)
+
+
+def test_make_batch_requests(small_mgf_path, sample_response):
```

Comment:
why do you use `sample_response` in this test?

Reply:
Unused. Forgot to remove